File: week7/윤여준/프린터.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    
    deque<pair<int,int>> dq;
    
    for(int i = 0; i < priorities.size(); i++)
        dq.push_back({i, priorities[i]});
    
    while(true) {
        pair<int,int> cur = dq.front();
        dq.pop_front();
        
        bool flag = false;
        for(auto pii : dq) {
            if(pii.second > cur.second) {
                dq.push_back(cur);
                flag = true;
                break;
            }
        }
        if(flag) continue;
        
        answer++;
        
        if(location == cur.first) break;
    }
    
    return answer;
}

File: week7/한지희/프린터.kt
class Printer {
    fun solution(priorities: IntArray, location: Int): Int {
        var answer = 0
        val documents = mutableListOf<Pair<Int, Int>>()

        priorities.forEachIndexed { index, priority ->
            documents.add(Pair(index, priority))
        }

        while (documents.isNotEmpty()) {
            val current = documents.first()
            documents.removeFirst()

            if (documents.any { it.second > current.second }) documents.add(current)
            else {
                answer++
                if (location == current.first) break
            }
        }

        return answer
    }
}